src/protocols/secure_channel/tests/TestCASESession.cpp

```diff
@@ -874,6 +866,217 @@ TEST_F(TestCASESession, Sigma1ParsingTest)
     TestSigma1Parsing(mem, bufferSize, Sigma1SessionIdTooBig);
 }
 
+TEST_F(TestCASESession, EncodeSigma1Test)
+{
+    System::PacketBufferHandle msg;
+    CASESession::EncodeSigma1Inputs encodeParams;
+
+    uint8_t random[32];
+    EXPECT_EQ(chip::Crypto::DRBG_get_bytes(&random[0], 32), CHIP_NO_ERROR);
+    encodeParams.initiatorRandom    = ByteSpan(random, sizeof(random));
+    encodeParams.initiatorSessionId = 7315;
+    uint8_t destinationId[32]       = { 0xDE, 0xAD };
+    encodeParams.destinationId      = ByteSpan(destinationId, sizeof(destinationId));
+
+    ReliableMessageProtocolConfig MRPConfig = GetDefaultMRPConfig();
+    encodeParams.initiatorMrpConfig         = &MRPConfig;
+
+    // EncodeSigma1 should fail when there is no public key
+    EXPECT_EQ(CHIP_ERROR_INCORRECT_STATE, EncodeSigma1(msg, encodeParams));
+
+    Crypto::P256Keypair * EphemeralKey = gDeviceOperationalKeystore.AllocateEphemeralKeypairForCASE();
+    ASSERT_NE(EphemeralKey, nullptr);
+    EXPECT_EQ(CHIP_NO_ERROR, EphemeralKey->Initialize(ECPKeyTarget::ECDH));
+    encodeParams.pEphPubKey = &EphemeralKey->Pubkey();
+
+    // Succeed when Public Key is provided
+    EXPECT_EQ(CHIP_NO_ERROR, EncodeSigma1(msg, encodeParams));
+
+    // Free the PacketBuffer
+    msg = nullptr;
```

Comment:
Split all sub-cases in nested scopes if they are not going to be separate full test cases.

Reply:
Done, declared a `System::PacketBufferHandle msg; `in each nested scope and removed all the `msg = nullptr;`